javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -709,14 +709,14 @@ public int indexOf(T element, int from) {
 
     @Override
     public Vector<T> insertAll(int index, Iterable<? extends T> elements) {
-        if ((index < 0) || (index > length())) {
-            throw new IndexOutOfBoundsException("insert(" + index + ", e) on Vector of length " + length());
-        } else {
```

Comment:
Is the benefit that much?

I think it is important to have a clean programming style for the reason of maintainability. We add guards that check the method arguments at the start of our method bodies.